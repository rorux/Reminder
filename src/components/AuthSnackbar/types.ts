export interface IAuthSnackbarProps {
  message: string;
  clearError: () => void;
}
