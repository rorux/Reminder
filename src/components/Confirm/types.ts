import React from 'react';

export type TConfirmProps = {
  open: boolean;
  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
  handler: (event: React.MouseEvent<HTMLElement>, id: string) => void;
  id: string;
};
