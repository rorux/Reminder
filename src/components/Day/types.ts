import { TRecord } from '@store/records/types';

export type TDayProps = {
  number: null | number;
  records: Array<TRecord>;
};

export type TStylesDay = {
  backgroundColor?: string;
  cursor?: 'pointer';
};
