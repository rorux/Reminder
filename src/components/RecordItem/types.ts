import React from 'react';
import { TRecordAccordion } from '@components/RecordsAccordion/types';

export type TRecordItemProps = {
  record: TRecordAccordion;
  handleChange: (panel: string) => (event: React.SyntheticEvent, isExpanded: boolean) => void;
  expanded: string | false;
  handleDelete: (event: React.MouseEvent<HTMLElement>, id: string) => void;
};
