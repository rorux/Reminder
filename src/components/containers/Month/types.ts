import { TRecord } from '@store/records/types';

export type TMonthProps = {
  year: number;
  month: number;
  recordList: Array<TRecord>;
};
