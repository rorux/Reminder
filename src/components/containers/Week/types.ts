import { TRecord } from '@store/records/types';

export type TWeekProps = {
  weekArray: Array<null | number>;
  recordList: Array<TRecord>;
};
