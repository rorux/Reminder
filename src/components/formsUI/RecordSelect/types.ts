export type TSelectParams = {
  value: string | number;
  name: string | number;
};

export type TSelectProps = {
  name: string;
  params: Array<TSelectParams>;
  setter: any;
  value?: any;
};
