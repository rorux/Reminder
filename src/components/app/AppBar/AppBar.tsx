import React, { useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useLocation } from 'react-router-dom';
import { styled } from '@mui/material/styles';
import MuiAppBar, { AppBarProps as MuiAppBarProps } from '@mui/material/AppBar';
import Typography from '@mui/material/Typography';
import MenuIcon from '@mui/icons-material/Menu';
import Toolbar from '@mui/material/Toolbar';
import IconButton from '@mui/material/IconButton';
import { Logout } from '@mui/icons-material';
import { appSelector } from '@store/app/selectors';
import { drawerToggleAction } from '@store/app/actions';
import { ROUTES, ROUTES_NAME } from '@router/types';
import { getEnumKeyByEnumValue, getEnumValueByEnumKey } from '@utils/funcs';
import { authLogoutAction } from '@store/auth/actions';

interface AppBarProps extends MuiAppBarProps {
  open?: boolean;
}

export const AppBarStyled = styled(MuiAppBar, {
  shouldForwardProp: (prop) => prop !== 'open',
})<AppBarProps>(({ theme, open }) => ({
  zIndex: theme.zIndex.drawer + 1,
  transition: theme.transitions.create(['width', 'margin'], {
    easing: theme.transitions.easing.sharp,
    duration: theme.transitions.duration.leavingScreen,
  }),
  ...(open && {
    marginLeft: 240,
    width: `calc(100% - 240px)`,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  }),
}));

export default function AppBar() {
  const { isDrawerOpen } = useSelector(appSelector);
  const dispatch = useDispatch();

  const location = useLocation().pathname;

  let title = '';

  if (location.substring(1, 5) === 'edit') {
    title = 'Редактирование записи';
  } else {
    const routeKey = getEnumKeyByEnumValue(ROUTES, location);
    title = getEnumValueByEnumKey(ROUTES_NAME, routeKey);
  }

  const toggleDrawer = useCallback(() => {
    dispatch(drawerToggleAction());
  }, [dispatch]);

  const handlerLogout = useCallback(() => {
    dispatch(authLogoutAction());
  }, [dispatch]);

  return (
    <AppBarStyled position="absolute" open={isDrawerOpen}>
      <Toolbar
        sx={{
          pr: '24px',
        }}
      >
        <IconButton
          data-test="toggle-drawer"
          edge="start"
          color="inherit"
          aria-label="open drawer"
          onClick={toggleDrawer}
          sx={{
            marginRight: '36px',
            ...(isDrawerOpen && { display: 'none' }),
          }}
        >
          <MenuIcon />
        </IconButton>
        <Typography component="h1" variant="h6" color="inherit" noWrap sx={{ flexGrow: 1 }}>
          {title}
        </Typography>
        <IconButton data-test="logout" aria-label="logout" onClick={handlerLogout}>
          <Logout sx={{ color: 'white' }} />
        </IconButton>
      </Toolbar>
    </AppBarStyled>
  );
}
