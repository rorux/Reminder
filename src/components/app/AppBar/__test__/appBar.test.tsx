import React from 'react';
import * as redux from 'react-redux';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { shallow } from 'enzyme';
import AppBar, { AppBarStyled } from '../AppBar';
import { AppActionTypes } from '@store/app/types';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: jest.fn(() => ({
    pathname: '/edit',
  })),
}));

describe('AppBar component', () => {
  let component: Enzyme.ShallowWrapper;
  let spyOnUseSelector;
  let spyOnUseDispatch;
  let mockDispatch: jest.Mock;

  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector');
    spyOnUseSelector.mockReturnValue({
      isDrawerOpen: false,
    });
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch');
    mockDispatch = jest.fn();
    spyOnUseDispatch.mockReturnValue(mockDispatch);
    component = shallow(<AppBar />);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('Should render AppBar component', () => {
    expect(component).toMatchSnapshot();
  });

  it('Should render AppBarStyled component', () => {
    expect(component.find(AppBarStyled)).toHaveLength(1);
  });

  it('Should dispatch drawerToggleAction when clicked toggle', () => {
    // @ts-ignore
    component.find(`[data-test='toggle-drawer']`).props().onClick();
    expect(mockDispatch.mock.calls[0][0]).toEqual({
      type: AppActionTypes.DRAWER_TOGGLE,
    });
  });

  it('Should dispatch authLogoutAction when clicked logout', () => {
    // @ts-ignore
    component.find(`[data-test='logout']`).props().onClick();
    expect(mockDispatch).toHaveBeenCalled();
  });
});

describe('AppBarStyled component', () => {
  it('Should render AppBarStyled component', () => {
    const component = shallow(<AppBarStyled />);
    expect(component).toMatchSnapshot();
  });
});
