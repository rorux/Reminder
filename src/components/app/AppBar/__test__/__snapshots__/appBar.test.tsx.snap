// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar component Should render AppBar component 1`] = `
<Styled(ForwardRef(AppBar))
  open={false}
  position="absolute"
>
  <ForwardRef(Toolbar)
    sx={
      Object {
        "pr": "24px",
      }
    }
  >
    <ForwardRef(IconButton)
      aria-label="open drawer"
      color="inherit"
      data-test="toggle-drawer"
      edge="start"
      onClick={[Function]}
      sx={
        Object {
          "marginRight": "36px",
        }
      }
    >
      <Memo(ForwardRef(MenuIcon)) />
    </ForwardRef(IconButton)>
    <ForwardRef(Typography)
      color="inherit"
      component="h1"
      noWrap={true}
      sx={
        Object {
          "flexGrow": 1,
        }
      }
      variant="h6"
    >
      Редактирование записи
    </ForwardRef(Typography)>
    <ForwardRef(IconButton)
      aria-label="logout"
      data-test="logout"
      onClick={[Function]}
    >
      <Memo(ForwardRef(LogoutIcon))
        sx={
          Object {
            "color": "white",
          }
        }
      />
    </ForwardRef(IconButton)>
  </ForwardRef(Toolbar)>
</Styled(ForwardRef(AppBar))>
`;

exports[`AppBarStyled component Should render AppBarStyled component 1`] = `
<ContextProvider
  value={
    Object {
      "insert": [Function],
      "inserted": Object {
        "1ryja0n": true,
      },
      "key": "css",
      "nonce": undefined,
      "registered": Object {
        "css-1ryja0n": "z-index:1201;transition:width 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms,margin 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;",
      },
      "sheet": StyleSheet {
        "_insertTag": [Function],
        "before": null,
        "container": undefined,
        "ctr": 0,
        "insertionPoint": undefined,
        "isSpeedy": false,
        "key": "css",
        "nonce": undefined,
        "prepend": undefined,
        "tags": Array [],
      },
    }
  }
>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1ryja0n{z-index:1201;-webkit-transition:width 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms,margin 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;transition:width 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms,margin 195ms cubic-bezier(0.4, 0, 0.6, 1) 0ms;}",
      }
    }
    data-emotion="css 1ryja0n"
  />
  <ForwardRef(AppBar)
    className="css-1ryja0n"
  />
</ContextProvider>
`;
